client/app/bundles/course/user-invitations/components/forms/IndividualInvitations.tsx

```diff
@@ -54,8 +78,70 @@ const IndividualInvitations: FC<Props> = (props) => {
     });
   };
 
+  const appendInputs = (results: InvitationEntry[]): void => {
+    const lastRow = fields[fields.length - 1];
+
+    if (fields.length === 1 && results.length > 0) {
+      remove(0);
+    }
+
+    results.forEach((entry) => {
+      append({
+        name: entry.name,
+        email: entry.email,
+        role: lastRow.role,
+        phantom: lastRow.phantom,
+        ...(permissions.canManagePersonalTimes && {
+          timelineAlgorithm: lastRow.timelineAlgorithm,
```

Comment:
Code duplication here. You should make use of `appendNewRow` and extend it by allowing `name` and `email` arguments (default `''`).

Reply:
done